web/client/reducers/timeline.js

```diff
@@ -74,14 +74,12 @@ module.exports = (state = {
             return set('mouseEvent', action.eventData, state);
         }
         case REMOVE_NODE: {
-            if (state.selectedLayer === action.node) {
-                let newState = state;
-                return assign({}, state, {
-                     rangeData: pickBy(newState.rangeData, (values, key) => key !== action.node),
-                     loading: pickBy(newState.loading, (values, key) => key !== action.node),
-                     selectedLayer: ''
+            let newState = state;
+            return assign({}, state, {
+                rangeData: has(newState.rangeData, action.node) && pickBy(newState.rangeData, (values, key) => key !== action.node),
```

Comment:
This is most probably a big error. In case of 
`has(newState.rangeData, action.node) == falsy` (so every time you delete a layer that has not time dimension) you reset the whole object.
The same for loading and `rangeData`, `loading` and `selectedLayer`. 

Please  fix this problem and write unit test and verify that : 
 - rangeData and loading remains the same in case you're removing a layer that is not in the rangeData or loading map, and that selectedLayer remains the same too.
 - the the other keys of rangeData and loading remains the same and only the proper key is removed. Selected layer is removed only if it was selected, otherwise it remains the same